api/client/src/main/java/org/projectnessie/client/http/Status.java

```diff
@@ -15,55 +15,132 @@
+  public static final Status REQUEST_URI_TOO_LONG =
+      addCode(REQUEST_URI_TOO_LONG_CODE, "Request-URI Too Long");
+  public static final int UNSUPPORTED_MEDIA_TYPE_CODE = 415;
+  public static final Status UNSUPPORTED_MEDIA_TYPE =
+      addCode(UNSUPPORTED_MEDIA_TYPE_CODE, "Unsupported Media Type");
+  public static final int REQUESTED_RANGE_NOT_SATISFIABLE_CODE = 416;
+  public static final Status REQUESTED_RANGE_NOT_SATISFIABLE =
+      addCode(REQUESTED_RANGE_NOT_SATISFIABLE_CODE, "Requested Range Not Satisfiable");
+  public static final int EXPECTATION_FAILED_CODE = 417;
+  public static final Status EXPECTATION_FAILED =
+      addCode(EXPECTATION_FAILED_CODE, "Expectation Failed");
+  public static final int PRECONDITION_REQUIRED_CODE = 428;
+  public static final Status PRECONDITION_REQUIRED =
+      addCode(PRECONDITION_REQUIRED_CODE, "Precondition Required");
+  public static final int TOO_MANY_REQUESTS_CODE = 429;
+  public static final Status TOO_MANY_REQUESTS =
+      addCode(TOO_MANY_REQUESTS_CODE, "Too Many Requests");
+  public static final int REQUEST_HEADER_FIELDS_TOO_LARGE_CODE = 431;
+  public static final Status REQUEST_HEADER_FIELDS_TOO_LARGE =
+      addCode(REQUEST_HEADER_FIELDS_TOO_LARGE_CODE, "Request Header Fields Too Large");
+  public static final int INTERNAL_SERVER_ERROR_CODE = 500;
+  public static final Status INTERNAL_SERVER_ERROR =
+      addCode(INTERNAL_SERVER_ERROR_CODE, "Internal Server Error");
+  public static final int NOT_IMPLEMENTED_CODE = 501;
+  public static final Status NOT_IMPLEMENTED = addCode(NOT_IMPLEMENTED_CODE, "Not Implemented");
+  public static final int BAD_GATEWAY_CODE = 502;
+  public static final Status BAD_GATEWAY = addCode(BAD_GATEWAY_CODE, "Bad Gateway");
+  public static final int SERVICE_UNAVAILABLE_CODE = 503;
+  public static final Status SERVICE_UNAVAILABLE =
+      addCode(SERVICE_UNAVAILABLE_CODE, "Service Unavailable");
+  public static final int GATEWAY_TIMEOUT_CODE = 504;
+  public static final Status GATEWAY_TIMEOUT = addCode(GATEWAY_TIMEOUT_CODE, "Gateway Timeout");
+  public static final int HTTP_VERSION_NOT_SUPPORTED_CODE = 505;
+  public static final Status HTTP_VERSION_NOT_SUPPORTED =
+      addCode(HTTP_VERSION_NOT_SUPPORTED_CODE, "HTTP Version Not Supported");
+  public static final int NETWORK_AUTHENTICATION_REQUIRED_CODE = 511;
+  public static final Status NETWORK_AUTHENTICATION_REQUIRED =
+      addCode(NETWORK_AUTHENTICATION_REQUIRED_CODE, "Network Authentication Required");
 
   private final int code;
   private final String reason;
 
-  Status(final int statusCode, final String reason) {
+  private static Status addCode(int code, String reason) {
+    Status[][] lines = LINES;
+    int line = code / 100;
+    int code100 = code % 100;
+    Status[] l = lines[line];
+    if (l.length < code100 + 1) {
+      l = lines[line] = Arrays.copyOf(l, code100 + 1);
+    }
+    return l[code100] = new Status(code, reason);
```

Comment:
`n % 100 + ( ((n / 100) - 2) * 10 + ((n / 100) - 2) / 3 * 22)` gives an non-overlapping index  between `0` and `63` (incl) for known codes ;)

Reply:
???

Reply:
You can use a small (one level) array with that formula for computing the index from the status code.